.github/workflows/matomo-tests.yml

```diff
@@ -0,0 +1,173 @@
+name: Matomo Tests
+
+on:
+  pull_request:
+
+permissions:
+  actions: read
+  checks: none
+  contents: read
+  deployments: none
+  issues: read
+  packages: none
+  pull-requests: read
+  repository-projects: none
+  security-events: none
+  statuses: none
+concurrency:
+  group: php-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  PHP:
+    runs-on: ubuntu-20.04
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis-sentinel:
+        image: 'bitnami/redis-sentinel:latest'
+        ports:
+          - 26379:26379
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_DATABASE: matomo_tests
+          MYSQL_ALLOW_EMPTY_PASSWORD: yes
+        options: --tmpfs /var/lib/mysql:rw
```

Comment:
should we define the required services in the github action as well? Only services that might differ from the default should be defined.

For plugins it would in general be awesome to have the action in each plugin as small as possible. That way we won't need to change every plugins action and only need to adjust our global test action if something needs to be adjusted.

Reply:
@sgiehl this may be a little bit tricky because Github Action composite does not accept containers. But I guess we can manually build MySQL and reds in ubuntu, instead of using docker.